pkg/watch/health.go

```diff
@@ -95,7 +89,7 @@ func MonitorPodHealth(config *preparer.PreparerConfig, logger *logging.Logger, s
 			// check if pods have been added or removed
 			// starts monitor routine for new pods
 			// kills monitor routine for removed pods
-			pods = updatePods(healthManager, client, pods, results, node, logger)
+			pods = updatePods(healthManager, config, pods, results, node, logger)
```

Comment:
This change will cause each new watch result to create a new `http.Client`, which is actually super dangerous. Each one caches pools of TCP connections. I'd rather you just init the insecure client outside this loop here and pass it in, both to prevent resource leaks as well as a package cleanliness point of view (we're trying to remove the `pkg/preparer` dependency in logically isolated packages, like health)